src/libstd/sys/unix/fs.rs

```diff
@@ -511,10 +511,19 @@ pub fn rmdir(p: &Path) -> io::Result<()> {
 }
 
 pub fn remove_dir_all(path: &Path) -> io::Result<()> {
+    let filetype = try!(lstat(path)).file_type();
+    if filetype.is_symlink() {
+        unlink(path)
+    } else {
+        remove_dir_all_recursive(path)
+    }
+}
```

Comment:
The Windows version is similar but a bit tricky. It uses `rmdir()` so it can only remove directory symlinks and not file symlinks. Unix just uses `unlink()`. But I don't mind if we make the distinction here or in `std::fs`.